### PR TITLE
Allow setting /dev/shm size in DockerBuildImage and DockerCreateContainer

### LINKED ISSUE #382: about support of --shm-size when building docker image with gradle plugin ?
Hi 

i'm used to build docker image with option --shm-size 
E.g : docker build  --shm-size =4g ...etc

Does --shm-size option is  supported when building docker image with docker-gradle-plugin ?
if yes , could you please provide an example how to use it  ?

Thanks in advance 

Best Regards,
Oussema

### PR DESCRIPTION
**Pull request by orzeh**
This should fix #382

### CODE REVIEW

File: src/functTest/groovy/com/bmuschko/gradle/docker/DockerWorkflowFunctionalTest.groovy [hunk 1]
```diff
@@ -494,6 +494,45 @@ class DockerWorkflowFunctionalTest extends AbstractFunctionalTest {
...
+                targetImageId { buildImage.getImageId() }
+                containerName = "${uniqueContainerName}"
+                shmSize = 128000
+                cmd = ['/bin/sh']
+            }
+
+            task inspectContainer(type: DockerInspectContainer) {
+                dependsOn createContainer
+                targetContainerId { createContainer.getContainerId() }
+            }
```
Comment (cdancy) [inline src/functTest/groovy/com/bmuschko/gradle/docker/DockerWorkflowFunctionalTest.groovy]: Can we just use `onNext` and do what you did for `ResponseHandler` here? I'd like to move away from that and perhaps with the next minor version bump we can do so.

Comment (orzeh) [inline src/functTest/groovy/com/bmuschko/gradle/docker/DockerWorkflowFunctionalTest.groovy]: ok, good idea.

File: src/main/groovy/com/bmuschko/gradle/docker/response/container/InspectContainerResponseHandler.groovy [hunk 2]
```diff
@@ -20,10 +20,11 @@ class InspectContainerResponseHandler implements ResponseHandler<String, Object>
...
-        def devices = container.hostConfig.@devices ? 
-                container.hostConfig.devices.collect { 
-                    "${it.pathOnHost}:${it.pathInContainer}:${it.cGroupPermissions}" 
+        def devices = container.hostConfig.@devices ?
+                container.hostConfig.devices.collect {
+                    "${it.pathOnHost}:${it.pathInContainer}:${it.cGroupPermissions}"
                 } : '[]'
         logger.quiet "Devices : $devices"
+        logger.quiet "ShmSize : $container.hostConfig.shmSize"
     }
```
Comment (cdancy) [inline src/main/groovy/com/bmuschko/gradle/docker/response/container/InspectContainerResponseHandler.groovy]: Can we remove the above changes? I'd like to just rip the ResponseHandler stuff out with the next release as we'll be going to 3.1.0.

Comment (orzeh) [inline src/main/groovy/com/bmuschko/gradle/docker/response/container/InspectContainerResponseHandler.groovy]: no problem :smile:

**Comment by cdancy** [comment]
LGTM. Merging now. Thanks @orzeh !!!

**Pull request merged**